src/js/helpers/Util.js

```diff
@@ -122,6 +122,19 @@ var Util = {
   isEmptyString: function (str) {
     return this.isString(str) && (str == null || str === "");
   },
+  isValidPort: function (value) {
+    if (value == null || this.isEmptyString(value)) {
+      return true;
+    }
+    var port = parseInt(value, 10);
```

Comment:
With parseInt you strip out invalid characters, which means you cannot check to them.
Consider a RegExp.